src/go/rpk/pkg/config/weak_test.go

```diff
@@ -0,0 +1,555 @@
+			}
+		})
+	}
+}
+
+func TestWeakInt(t *testing.T) {
+	type testWeakInt struct {
+		Wi weakInt `yaml:"wi"`
+	}
+	for _, test := range []struct {
+		name   string
+		data   string
+		expInt int
+		expErr bool
+	}{
+		{
+			name:   "normal int types",
+			data:   "wi: 1231",
+			expInt: 1231,
+		},
+		{
+			name:   "empty string as 0",
+			data:   `wi: ""`,
+			expInt: 0,
+		},
+		{
+			name:   "string:-23414",
+			data:   `wi: "-23414"`,
+			expInt: -23414,
+		},
+		{
+			name:   "string:231231",
+			data:   `wi: "231231"`,
+			expInt: 231231,
+		},
+		{
+			name:   "bool:true",
+			data:   `wi: true`,
+			expInt: 1,
+		},
+		{
+			name:   "bool:false",
+			data:   `wi: false`,
+			expInt: 0,
+		},
+		{
+			name:   "error with non-numeric strings",
+			data:   `wi: "123foo234"`,
+			expErr: true,
+		},
+		{
+			name:   "error with float numbers",
+			data:   `wi: 123.234`,
+			expErr: true,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			ts := testWeakInt{}
+			err := yaml.Unmarshal([]byte(test.data), &ts)
+
+			gotErr := err != nil
+			if gotErr != test.expErr {
+				t.Errorf("input %q: got err? %v, exp err? %v; error: %v",
+					test.data, gotErr, test.expErr, err)
+				return
+			}
+			if test.expErr {
+				return
+			}
+
+			if int(ts.Wi) != test.expInt {
+				t.Errorf("input %q: got %v, expected %v",
+					test.data, ts.Wi, test.expInt)
+				return
+			}
+		})
+	}
+}
+
+func TestWeakString(t *testing.T) {
+	type testWeakString struct {
+		Ws weakString `yaml:"ws"`
+	}
+	for _, test := range []struct {
+		name   string
+		data   string
+		expStr string
+		expErr bool
+	}{
+		{
+			name:   "normal string",
+			data:   `ws: "hello world"`,
+			expStr: "hello world",
+		},
+		{
+			name:   "bool:true",
+			data:   "ws: true",
+			expStr: "1",
+		},
+		{
+			name:   "bool:false",
+			data:   "ws: false",
+			expStr: "0",
+		},
+		{
+			name:   "base10 number",
+			data:   "ws: 231231",
+			expStr: "231231",
+		},
+		{
+			name:   "float number",
+			data:   "ws: 231.231",
+			expStr: "231.231",
+		},
+		{
+			name:   "should error with unsupported types",
+			data:   "ws: \n  - 231.231",
+			expErr: true,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			ts := testWeakString{}
+			err := yaml.Unmarshal([]byte(test.data), &ts)
+
+			gotErr := err != nil
+			if gotErr != test.expErr {
+				t.Errorf("input %q: got err? %v, exp err? %v; error: %v",
+					test.data, gotErr, test.expErr, err)
+				return
+			}
+			if test.expErr {
+				return
+			}
+
+			if string(ts.Ws) != test.expStr {
+				t.Errorf("input %q: got %v, expected %v",
+					test.data, ts.Ws, test.expStr)
+				return
+			}
+		})
+	}
+}
+
+func TestWeakStringArray(t *testing.T) {
+	type testWeakStringArray struct {
+		WsArr weakStringArray `yaml:"test_array"`
+	}
+	for _, test := range []struct {
+		name   string
+		data   string
+		expArr []string
+		expErr bool
+	}{
+		{
+			name: "single weak string",
+			data: `test_array:
+  12`,
+			expArr: []string{"12"},
+		},
```

Comment:
Can we add two inline tests here, just to be sure? `test_array: 12` and `test_array: [12, true]`?